src/fe-common/core/fe-common-core.c

```diff
@@ -470,16 +470,23 @@ gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 		return TRUE;
 
 	if (dest->server_tag != NULL) {
-		char *tagtarget = g_strdup_printf("%s/%s", dest->server_tag, "*");
+		char *tagtarget = g_strdup_printf("%s/%s", dest->server_tag, "#");
```

Comment:
All the possible `tagtarget` values are in the form `server_tag/{#,@,*,target}` so it might make sense to have a `strarray_find_prefix` function so that you find the first entry matching `server_tag/` (the first one is the winner in case of duplicates) and then parse the part after the slash.
(please drop the previous commit and rebase this against master)